src/api/user/user.controller.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Post,
+  Put,
+  Query,
+} from '@nestjs/common';
+import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { UserService } from './user.service';
+import { CreateUserDto, UpdateUserDto, UserDto } from './dto/user.dto';
+import { MessageDto } from 'src/shared/dto/message.dto';
+import { PaginatedResult, PaginationDto } from 'src/shared/dto/pagination.dto';
+
+@ApiTags('User')
+@Controller('user')
+export class UserController {
```

Comment:
1. Your API is not following REST principles. See https://restfulapi.net/resource-naming/
2. You need some king of authentication(I suggest JWT) to protect endpoints like getting or deleting user. I would suggest endpoints POST /sign-up for registration, POST /sign-in for login, GET /users/current for gettings user info(it should be protected with JWT token and you will get user id from JWT claims), GET /users for getting all user info(it should be available only for admins, so you need roles for users, I would suggest skipping this part for know), PUT /users/current to update current user info(also JWT authentication), DELETE /users/current(also JWT)